File: binary-tree-inorder-traversal/binary-tree-inorder-traversal.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */


/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
void insert(struct TreeNode *root, int *array, int* size){
    if (root == NULL){
        return;
    }
    
    insert(root->left, array, size);
    array[(*size)++] = root->val;
    insert(root->right, array, size);
}

int* inorderTraversal(struct TreeNode* root, int* returnSize){
    int *array = malloc(sizeof(int) * 100);
    int *size = 0;
    
    insert(root, array, &size);
    *returnSize = size;

    return array;
    
    
    
}

File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        stack, arr = [], []
        curr = root
        while curr != None or len(stack) != 0:
            while curr != None:
                stack.append(curr)
                curr = curr.left
            curr = stack.pop()
            arr.append(curr.val)
            curr = curr.right
            
        return arr